pkg/vsock/handshake_windows.go

```diff
@@ -0,0 +1,123 @@
+package vsock
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	"github.com/Microsoft/go-winio"
+	"github.com/linuxkit/virtsock/pkg/hvsock"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/windows/registry"
+)
+
+// GetVMGUID retrieves the GUID for a correct hyper-v VM (most likely WSL).
+// It performs a handshake with a running vsock-peer in the WSL distro
+// to make sure we establish the AF_VSOCK connection with a right VM.
+func GetVMGUID(seed string, handshakePort uint32, timeout <-chan time.Time) (hvsock.GUID, error) {
+	key, err := registry.OpenKey(
+		registry.LOCAL_MACHINE,
+		`SOFTWARE\Microsoft\Windows NT\CurrentVersion\HostComputeService\VolatileStore\ComputeSystem`,
+		registry.ENUMERATE_SUB_KEYS)
+	if err != nil {
+		return hvsock.GUIDZero, fmt.Errorf("could not open registry key, is WSL VM running? %v", err)
+	}
+	defer key.Close()
+
+	names, err := key.ReadSubKeyNames(0)
+	if err != nil {
+		return hvsock.GUIDZero, fmt.Errorf("machine IDs can not be read in registry: %v", err)
+	}
+	if len(names) == 0 {
+		return hvsock.GUIDZero, errors.New("no running WSL VM found")
+	}
```

Comment:
(We can close `key` at this point)

Reply:
There is a `defer key.Close()` further up.